core/java/iesi-rest-without-microservices/src/main/java/io/metadew/iesi/server/rest/component/ComponentDtoRepository.java

```diff
@@ -0,0 +1,179 @@
+                "LEFT OUTER JOIN " + MetadataTablesConfiguration.getInstance().getMetadataTableNameByLabel("ComponentParameters").getName() + " parameters " +
+                "on base_components.COMP_ID = parameters.COMP_ID AND base_components.COMP_VRS_NB = parameters.COMP_VRS_NB " +
+                getOrderByClause(pageable) +
+                ";";
+    }
+
+    private String getBaseQuery(Authentication authentication, Pageable pageable, List<ComponentFilter> componentFilters) {
+        return "select distinct component_designs.COMP_ID, versions.COMP_VRS_NB " +
+                "from " + MetadataTablesConfiguration.getInstance().getMetadataTableNameByLabel("Components").getName() + " component_designs " +
+                "INNER JOIN " + MetadataTablesConfiguration.getInstance().getMetadataTableNameByLabel("ComponentVersions").getName() + " versions " +
+                "on component_designs.COMP_ID = versions.COMP_ID" +
+                getWhereClause(componentFilters) +
+                getOrderByClause(pageable) +
+                getLimitAndOffsetClause(pageable);
+    }
+
+    @Override
+    public Page<ComponentDto> getAll(Authentication authentication, Pageable pageable, List<ComponentFilter> componentFilters) {
+        try {
+            Map<ComponentKey, ComponentDtoBuilder> componentDtoBuilders = new LinkedHashMap<>();
+            String query = getFetchAllQuery(authentication, pageable, componentFilters);
+            CachedRowSet cachedRowSet = metadataRepositoryConfiguration.getDesignMetadataRepository().executeQuery(query, "reader");
+
+            while (cachedRowSet.next()) {
+                mapRow(cachedRowSet, componentDtoBuilders);
+            }
+            List<ComponentDto> componentDtoList = componentDtoBuilders.values().stream()
+                    .map(ComponentDtoBuilder::build)
+                    .collect(Collectors.toList());
+            return new PageImpl<>(componentDtoList, pageable, getRowSize(authentication));
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private long getRowSize(Authentication authentication) throws SQLException {
+        String query = "select count(*) as row_count from (select distinct component_designs.COMP_ID, versions.COMP_VRS_NB " +
+                "from " + MetadataTablesConfiguration.getInstance().getMetadataTableNameByLabel("Components").getName() + " component_designs " +
+                "INNER JOIN " + MetadataTablesConfiguration.getInstance().getMetadataTableNameByLabel("ComponentVersions").getName() + " versions " +
+                "on component_designs.COMP_ID = versions.COMP_ID );";
+        CachedRowSet cachedRowSet = metadataRepositoryConfiguration.getDesignMetadataRepository().executeQuery(query, "reader");
+        cachedRowSet.next();
+        return cachedRowSet.getLong("row_count");
```

Comment:
Are you already using filters?
If so, shouldn't they be included in the 'count' query?